src/query/procedure/py_module.cpp

```diff
@@ -905,30 +918,66 @@ std::optional<py::ExceptionInfo> AddRecordFromPython(mgp_result *result, py::Obj
       PyErr_SetString(PyExc_TypeError, msg.c_str());
       return py::FetchError();
```

Comment:
If we return here py::FetchError whole std::vector will be still cached and we have a leak. I think we need to add here onScopeExit

Reply:
For std::vector

Reply:
Added